fapolicy_analyzer/ui/subject_list.py

```diff
@@ -164,12 +197,41 @@ def on_view_row_activated(self, treeView, row, *args):
         self.__show_reconciliation_dialog(subject)
 
     def on_view_button_press_event(self, widget, event):
+        treeView = self.get_object("treeView")
+        model, pathlist = treeView.get_selection().get_selected_rows()
+        n_paths = len(pathlist)
         if event.type == Gdk.EventType.BUTTON_PRESS and event.button == 3:
-            self.contextMenu.popup_at_pointer()
+            if n_paths == 1:
+                self.reconcileContextMenu.popup_at_pointer()
+            elif n_paths > 1:
+                trust = list(
+                    {
+                        self.find_db_trust(model.get_value(model.get_iter(path), 3))
+                        for path in iter(pathlist)
+                    }
+                )
```

Comment:
This logic is harder than I was first thinking, and this isn't going to work.  You need to but some logic in to determine the  list of actions (trust or untrust) based on both the `subject.trust` (AT, ST, U) and the `dbtrust.status` (T, D).  Then check if all the actions are the same.  If not, exit out without giving the user to any menu options.